client-reactjs/src/components/application/QueryDetails.js

```diff
@@ -459,10 +497,46 @@ class QueryDetails extends PureComponent {
     };
     const selectedCluster = clusters.filter(cluster => cluster.id == this.props.clusterId);
 
+      //Function to make fields editable
```

Comment:
Should these two functions be moved to some kind of a utility class since its being used in multiple components?

Reply:
Yes I was thinking of re-factoring it